wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wom/graph/CallElementToGraphNode.scala

```diff
@@ -2,34 +2,126 @@ package wdl.draft3.transforms.wdlom2wom.graph
+            callInputPorts = Set(callNodeBuilder.makeInputPort(inputDefinition, expressionNode.singleOutputPort)),
+            newExpressionNodes = Set(expressionNode)
+          )
+
+        // No input mapping, use the default expression
+        case withDefault@InputDefinitionWithDefault(_, _, expression, _) =>
+          InputDefinitionFold(
+            mappings = List(withDefault -> Coproduct[InputDefinitionPointer](expression))
+          )
+
+        // No input mapping, required and we don't have a default value, create a new RequiredGraphInputNode
+        // so that it can be satisfied via workflow inputs
+        case required@RequiredInputDefinition(n, womType, _) =>
+          val identifier = WomIdentifier(n.value)
+          withGraphInputNode(required, RequiredGraphInputNode(identifier, womType, identifier.fullyQualifiedName.value))
+
+        // No input mapping, no default value but optional, create a OptionalGraphInputNode
+        // so that it can be satisfied via workflow inputs
+        case optional@OptionalInputDefinition(n, womType, _) =>
+          val identifier = WomIdentifier(n.value)
+          withGraphInputNode(optional, OptionalGraphInputNode(identifier, womType, identifier.fullyQualifiedName.value))
+      }
     }
 
-    callable map { callable =>
-      new CallNodeBuilder().build(WomIdentifier(a.node.callableName), callable, InputDefinitionFold(), true).nodes
+    (a.callableValidation, expressionNodeMappings) flatMapN { (callable, mappings) =>
+
+      val graphNodeSetter = new GraphNodeSetter[CallNode]
+      val result = callNodeBuilder.build(WomIdentifier(a.node.callableName), callable, foldInputDefinitions(mappings, callable))
+      graphNodeSetter._graphNode = result.node
+      result.nodes.valid
     }
   }
 }
 
-final case class CallableNodeMakerInputs(node: CallElement,
-                                         linkableValues: Map[UnlinkedConsumedValueHook, GeneratedValueHandle],
-                                         linkablePorts: Map[String, Any],
-                                         availableTypeAliases: Map[String, WomType],
-                                         workflowName: String,
-                                         insideAnotherScatter: Boolean,
-                                         callables: Set[Callable])
\ No newline at end of file
+case class CallNodeMakerInputs(node: CallElement,
+                               linkableValues: Map[UnlinkedConsumedValueHook, GeneratedValueHandle],
+                               linkablePorts: Map[String, OutputPort],
+                               availableTypeAliases: Map[String, WomType],
+                               workflowName: String,
+                               insideAnotherScatter: Boolean,
+                               callables: Set[Callable]) {
+  // match the call element to a callable
+  val callableValidation: ErrorOr[Callable] = callables.find(_.name == node.callableName) match {
```

Comment:
I'd suggest moving this into the `CallElementToGraphNode.convert` method, or make it a public method in `CallElementToGraphNode` since I ended up using it in my small PR and we won't have access to a `CallNodeMakerInputs` there